### PR TITLE
add xls file suport

### PR DESCRIPTION
**Pull request by ic-xu**
# Description

We found that files in xls format were not supported for uploading to the knowledge base, so we started writing a compatible branch code, hoping to contribute to the community.

## Type of Change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Directly upload the xls file for analysis

### CODE REVIEW

**Comment by crazywoola** [comment]
@Weaxs Please take a look at this as well.

File: api/core/rag/extractor/excel_extractor.py [hunk 1]
```diff
@@ -43,5 +71,18 @@ def extract(self) -> list[Document]:
                 item = ';'.join(f'{k}:{v}' for k, v in row.items() if pd.notna(v))
                 document = Document(page_content=item, metadata={'source': self._file_path})
                 data.append(document)
-
         return data
+
+    @staticmethod
+    def is_blank_row(row):
+        """
+        判断指定行是否为空白行。
```
Comment (crazywoola) [inline api/core/rag/extractor/excel_extractor.py]: Please use English in code.

File: web/yarn.lock
Comment (crazywoola) [inline web/yarn.lock:1]: Please revert this file since there are no changes in frontend part.

**Review by crazywoola — CHANGES_REQUESTED** [review]
Hello thank you for your contribution, you can add my wechat: crazyphage
I will invite you to our contributor's group.

File: api/core/rag/extractor/excel_extractor.py [hunk 2]
```diff
@@ -27,10 +28,37 @@ def __init__(
...
     def extract(self) -> list[Document]:
+        """ parse excel file"""
+        if self._file_path.endswith('.xls'):
+            return self._extract4xls()
+        elif self._file_path.endswith('.xlsx'):
+            return self._extract4xlsx()
+
+    def _extract4xls(self) -> list[Document]:
+        wb = xlrd.open_workbook(filename=self._file_path)
+        row_header = None
```
Comment (Weaxs) [inline api/core/rag/extractor/excel_extractor.py]: every sheet needs to initialize  row_header， maybe move this to line 43 after `for sheet in wb.sheets():`

File: api/core/rag/extractor/excel_extractor.py [hunk 3]
```diff
@@ -27,10 +28,37 @@ def __init__(
...
+                documents.append(document)
+        return documents
+
+    def _extract4xlsx(self) -> list[Document]:
         """Load from file path using Pandas."""
         data = []
-
-        # 使用 Pandas 读取 Excel 文件的每个工作表
+        # Read each worksheet of an Excel file using Pandas
         xls = pd.ExcelFile(self._file_path)
```
Comment (Weaxs) [inline api/core/rag/extractor/excel_extractor.py:62]: `pd.ExcelFile(self._file_path)` can read xlsx and xls, It's not necessary to add functions `_extract4xls` and `_extract4xlsx` ?

**Review by Weaxs — COMMENTED** [review]
It's not necessary to change `excel_extractor.py`, just add `.xls'` type in `extract_processor.py`  and `file_service.py`

File: web/yarn.lock
Comment (bowenliang123) [inline web/yarn.lock:1]: leave it untouched, since no npm dependencies updated.

File: api/core/rag/extractor/excel_extractor.py (hunk 2 above)
Comment (ic-xu) [reply api/core/rag/extractor/excel_extractor.py]: I'm sorry for not thinking of this part, he did have such logic, I will fix it

File: api/core/rag/extractor/excel_extractor.py (hunk 3 above)
Comment (ic-xu) [reply api/core/rag/extractor/excel_extractor.py:62]: Yes, we need it. Pandas cannot read media information. We may need to read media information later.

**Review by crazywoola — APPROVED** [review]

**Pull request merged**